packages/core/src/blocks/CodeBlockContent/CodeBlockContent.ts

```diff
@@ -0,0 +1,359 @@
+import { InputRule, isTextSelection } from "@tiptap/core";
+import {
+  BuiltinLanguage,
+  bundledLanguagesInfo,
+  createHighlighter,
+  Highlighter,
+} from "shiki";
+import {
+  PropSchema,
+  createBlockSpecFromStronglyTypedTiptapNode,
+  createStronglyTypedTiptapNode,
+} from "../../schema/index.js";
+import { defaultProps } from "../defaultProps.js";
+import { createHighlightPlugin, Parser } from "prosemirror-highlight";
+import { createParser } from "prosemirror-highlight/shiki";
+import { TextSelection } from "@tiptap/pm/state";
+import { createDefaultBlockDOMOutputSpec } from "../defaultBlockHelpers.js";
+
+type SupportedLanguageConfig = {
+  id: string;
+  name: string;
+  match: string[];
+};
+
+export const codeBlockPropSchema = {
+  ...defaultProps,
```

Comment:
@YousefED Should the code block support default props (e.g. `textAlignment`, `backgroundColor`, etc.). I've referenced it from other blocks, but I'm not sure if it makes sense here

Reply:
good spot, let's remove